src/components/BarChart.js

```diff
@@ -1,299 +1,351 @@
 const React = require('react');
-const numeral = require('numeral');
 const _isEqual = require('lodash/isEqual');
+const _merge = require('lodash/merge');
+const _omit = require('lodash/omit');
+const _functions = require('lodash/functions');
 
 const StyleConstants = require('../constants/Style');
 
-const SetIntervalMixin = {
-  componentWillMount () {
-    this.intervals = [];
-  },
-  setInterval () {
-    this.intervals.push(setInterval.apply(null, arguments));
-  },
-  componentWillUnmount () {
-    this.intervals.map(clearInterval);
-  }
-};
-
-const Rect = React.createClass({
+const Bar = React.createClass({
   propTypes: {
     animateOnHover: React.PropTypes.bool,
-    color: React.PropTypes.string,
-    height: React.PropTypes.number.isRequired,
-    label: React.PropTypes.string,
+    animationDuration: React.PropTypes.number,
+    height: React.PropTypes.number,
+    hovering: React.PropTypes.bool,
     onClick: React.PropTypes.func,
     onMouseOut: React.PropTypes.func,
     onMouseOver: React.PropTypes.func,
-    primaryColor: React.PropTypes.string,
-    value: React.PropTypes.number.isRequired,
+    radius: React.PropTypes.number,
+    value: React.PropTypes.number,
     width: React.PropTypes.number,
-    x: React.PropTypes.number.isRequired,
-    y: React.PropTypes.number.isRequired
-  },
-
-  mixins: [SetIntervalMixin],
-
-  getInitialState () {
-    return {
-      hovering: false,
-      milliseconds: 0
-    };
+    x: React.PropTypes.number,
+    y: React.PropTypes.number
   },
 
   componentDidMount () {
-    this.setInterval(this.tick, 10);
+    if (this.props.animationDuration) {
+      const transform = this._getTransformWithScale(0);
+
+      d3.select(this.bar)
+        .attr('transform', transform)
+      .transition()
+        .duration(this.props.animationDuration)
+        .attr('transform', 'scale(1)');
+    }
   },
 
-  componentWillReceiveProps (nextProps) {
-    if (nextProps.value !== this.props.value) {
-      this.setState({
-        milliseconds: 0
-      });
-    }
+  shouldComponentUpdate (nextProps) {
+    // check the props that aren't functions
+    return !_isEqual(_omit(nextProps, _functions(nextProps)), _omit(this.props, _functions(this.props)));
   },
 
-  tick () {
-    this.setState({
-      milliseconds: this.state.milliseconds + 10
-    });
+  componentDidUpdate () {
+    if (this.props.hovering && this.props.animateOnHover) {
+      const transform = this._getTransformWithScale(0.9);
+
+      d3.select(this.bar)
+        .transition()
+          .duration(100)
+          .attr('transform', transform)
+        .transition()
+          .duration(100)
+          .attr('transform', 'scale(1)');
+    }
   },
 
-  _handleMouseOver (label, value, x, y) {
-    const animateDuration = 500;
+  _getTransformWithScale (factor) {
+    const centerX = 0;
+    const centerY = (this.props.value > 0) ? this.props.y + this.props.height : this.props.y;
```

Comment:
It has to do with the direction of the transform that needs to happen for the intro animation and the bounce animation.

![ezgif com-video-to-gif 15](https://cloud.githubusercontent.com/assets/758371/22380974/d5e6557c-e47c-11e6-96d0-a99d21f15a94.gif)